### DISCUSSION TITLE
Pre-Processor define code block dimming or highlighting?

### DISCUSSION THREAD

**Discussion by EnviralDesign** [google/filament#6636] [General]
Hi, apologies if this is not a topic to chat about here as it's not directly related to filament, but I am asking because of my time working with filament - Is there a recommended IDE or tool or extension etc, preferably vscode or visual studio that can properly fold and / or dim code blocks that are deemed inactive within pre processor defines?

These features work fine out of the box in all major IDE's for c++, but cannot for the life of me figure out how to do this with straight glsl code, and the filament generated glsl is full of pre processor conditionals, so it would help a lot to be able to visualize them better.

Thank you!

**Comment by romainguy** [comment]
I don't know of a GLSL code editor that does this, no.

**Reply by EnviralDesign** [reply]
thanks for the input, I am getting around it for now changing the file association of glsl to cpp, and it changes up the highlighting but seems harmless enough and get the dimming. so will proceed with that for now.
